### PR TITLE
[TF] Fix accuracy issues in a couple of notebooks

### PR DESCRIPTION
**Pull request by terrytangyuan**
*Description of changes:*

This PR should fix the accuracy issues in the following sections:
* Softmax regression concise implementation
* Dropout
* MLP concise

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch.md [hunk 1]
```diff
@@ -432,7 +432,7 @@ def train_epoch_ch3(net, train_iter, loss, updater):  #@save
 
 ```{.python .input}
 #@tab tensorflow
-def train_epoch_ch3(net, train_iter, loss, updater):  #@save
+def train_epoch_ch3(net, train_iter, loss, updater, params=None):  #@save
```
Comment (mli) [inline chapter_linear-networks/softmax-regression-scratch.md:435]: why we need `params`? can we remove it to make the api consistent.

Comment (terrytangyuan) [reply chapter_linear-networks/softmax-regression-scratch.md:435]: In dropout section, net is a custom function which doesn’t have trainable_variables and its optimizer/updater is from tf.keras instead of using the Updater class you created in previous section. We need to pass params since we cannot attach it to either the optimizer or the custom model function.

Comment (mli) [reply chapter_linear-networks/softmax-regression-scratch.md:435]: Can you use the `Updater` class instead?

Comment (terrytangyuan) [reply chapter_linear-networks/softmax-regression-scratch.md]: Okay I switched to use `Updater` here. I originally thought that since the PyTorch implementation here uses `torch.optim.SGD` so it might be better to support `tf.keras.Optimizer` in `train_ch3()`.

**Comment by mli** [comment]
Thanks
